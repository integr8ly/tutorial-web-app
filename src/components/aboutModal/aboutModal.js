import React from 'react';
import PropTypes from 'prop-types';

import { AboutModal as PfAboutModal, TextContent, TextList, TextListItem } from '@patternfly/react-core';

import { detect } from 'detect-browser';
import redHatLogo from '../../img/Logo-RedHat-A-Reverse-RGB.svg';
import managedIntegrationLogo from '../../img/Logo-Red_Hat-Managed_Integration-A-Reverse-RGB.svg';
import pfBackgroundImage from '../../img/PF4DownstreamBG.svg';

const pkgJson = require('../../../package.json');

class AboutModal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  getLogo = () => {
    let clusterType = '';
    let logoName = '';
    if (window.OPENSHIFT_CONFIG) {
      clusterType = window.OPENSHIFT_CONFIG.mockData ? 'localhost' : window.OPENSHIFT_CONFIG.clusterType;
      if (clusterType === 'poc') {
        logoName = managedIntegrationLogo;
      } else if (clusterType === 'osd') {
        logoName = managedIntegrationLogo;
      } else {
        logoName = redHatLogo;
      }
    }
    return logoName;
  };

  render() {
    const { isOpen, closeAboutModal } = this.props;
    const browser = detect();

    const urlParts = window.location.host.split('.');
    const [, , clusterId] = urlParts;

    return (
      <React.Fragment>
        <PfAboutModal
          isOpen={isOpen}
          onClose={closeAboutModal}
          productName="Red Hat Solution Explorer"
          brandImageSrc={this.getLogo()}
          brandImageAlt="Red Hat logo"
          backgroundImageSrc={pfBackgroundImage}
        >
          <TextContent>
            <TextList component="dl">
              <TextListItem component="dt">Integreatly Version</TextListItem>
              <TextListItem component="dd">
                {window.OPENSHIFT_CONFIG ? window.OPENSHIFT_CONFIG.integreatlyVersion : ' '}
              </TextListItem>
              <TextListItem component="dt">Console Version</TextListItem>
              <TextListItem component="dd">{pkgJson.version}</TextListItem>
              <TextListItem component="dt">Cluster Name</TextListItem>
              <TextListItem component="dd">{clusterId}</TextListItem>
              <TextListItem component="dt">User Name</TextListItem>
              <TextListItem component="dd">{window.localStorage.getItem('currentUserName')}</TextListItem>
              <TextListItem component="dt">Browser Version</TextListItem>
              <TextListItem component="dd">
                {browser ? browser.name : ' '} {browser ? browser.version : ' '}
              </TextListItem>
              <TextListItem component="dt">Browser OS</TextListItem>
              <TextListItem component="dd">{browser ? browser.os : ' '}</TextListItem>
            </TextList>
          </TextContent>
        </PfAboutModal>
      </React.Fragment>
    );
  }
}

AboutModal.propTypes = {
  isOpen: PropTypes.bool,
  closeAboutModal: PropTypes.func
};

AboutModal.defaultProps = {
  isOpen: false,
  closeAboutModal: null
};

export { AboutModal as default, AboutModal };
