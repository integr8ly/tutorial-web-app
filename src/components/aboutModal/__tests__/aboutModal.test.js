import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { ConnectedAboutModal, AboutModal } from '../aboutModal';

describe('AboutModal Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component with default props', () => {
    const store = generateEmptyStore({ aboutModal: { show: true }, user: { session: { username: 'test' } } });
    const component = shallow(<ConnectedAboutModal />, { context: { store } });

    expect(component).toMatchSnapshot('connected');
  });

  it('should render a non-connected component', () => {
    const props = {
      show: false,
      user: { session: { username: 'test' } }
    };

    const component = mount(<AboutModal {...props} />);
    expect(component).toMatchSnapshot('hidden modal');

    component.setState({ show: true });
    expect(component).toMatchSnapshot('show modal');
  });
});
