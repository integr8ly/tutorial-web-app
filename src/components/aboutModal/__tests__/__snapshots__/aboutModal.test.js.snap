// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutModal Component should render a non-connected component: hidden modal 1`] = `
<AboutModal
  closeAboutModal={null}
  isOpen={false}
  show={false}
  userReducers={
    Object {
      "session": Object {
        "username": "test",
      },
    }
  }
>
  <AboutModal
    appendTo={null}
    backgroundImageSrc="PF4DownstreamBG.svg"
    brandImageAlt="Red Hat logo"
    brandImageSrc=""
    className=""
    isOpen={false}
    noAboutModalBoxContentContainer={false}
    onClose={null}
    productName="Red Hat Solution Explorer"
    trademark=""
  >
    <Portal
      containerInfo={<div />}
    >
      <AboutModalContainer
        ariaDescribedById="pf-about-modal-content-0"
        ariaLabelledbyId="pf-about-modal-title-0"
        backgroundImageSrc="PF4DownstreamBG.svg"
        brandImageAlt="Red Hat logo"
        brandImageSrc=""
        className=""
        isOpen={false}
        noAboutModalBoxContentContainer={false}
        onClose={null}
        productName="Red Hat Solution Explorer"
        trademark=""
      />
    </Portal>
  </AboutModal>
</AboutModal>
`;

exports[`AboutModal Component should render a non-connected component: show modal 1`] = `
<AboutModal
  closeAboutModal={null}
  isOpen={false}
  show={false}
  userReducers={
    Object {
      "session": Object {
        "username": "test",
      },
    }
  }
>
  <AboutModal
    appendTo={null}
    backgroundImageSrc="PF4DownstreamBG.svg"
    brandImageAlt="Red Hat logo"
    brandImageSrc=""
    className=""
    isOpen={false}
    noAboutModalBoxContentContainer={false}
    onClose={null}
    productName="Red Hat Solution Explorer"
    trademark=""
  >
    <Portal
      containerInfo={<div />}
    >
      <AboutModalContainer
        ariaDescribedById="pf-about-modal-content-0"
        ariaLabelledbyId="pf-about-modal-title-0"
        backgroundImageSrc="PF4DownstreamBG.svg"
        brandImageAlt="Red Hat logo"
        brandImageSrc=""
        className=""
        isOpen={false}
        noAboutModalBoxContentContainer={false}
        onClose={null}
        productName="Red Hat Solution Explorer"
        trademark=""
      />
    </Portal>
  </AboutModal>
</AboutModal>
`;
